apis/projectcontour/v1alpha1/contourconfig.go

```diff
@@ -55,14 +53,13 @@ type ContourConfigurationSpec struct {
 
 	// HTTPProxy defines parameters on HTTPProxy.
 	// +optional
-	// +kubebuilder:default={disablePermitInsecure: false}
-	HTTPProxy HTTPProxyConfig `json:"httpproxy"`
+	HTTPProxy *HTTPProxyConfig `json:"httpproxy,omitempty"`
 
 	// EnableExternalNameService allows processing of ExternalNameServices
-	// Defaults to disabled for security reasons.
+	//
+	// Contour's default is false for security reasons.
 	// +optional
-	// +kubebuilder:default=false
-	EnableExternalNameService bool `json:"enableExternalNameService"`
+	EnableExternalNameService *bool `json:"enableExternalNameService,omitempty"`
```

Comment:
I ended up making all boolean fields pointers. The reasoning is that, if we ever wanted to change the internal default for one of these to `true`, then if this weren't a pointer we'd have no way to distinguish between "user didn't provide a value for the field" and "user explicitly specified false". However, we do generally follow a pattern of boolean config fields always defaulting to false, so I'm definitely open to leaving these as non-pointers that always must default to false (it makes them easier to work with in code, fewer derefs). Looking for input.

More generally, my thinking with primitive types is - if the zero value is a valid option for the field, then the field probably should be a pointer to distinguish between "not specified" and "explicitly specified as the zero value". If the zero value is invalid, then the field doesn't need a pointer since we can assume anything with the zero value was not specified.